posts/google-summer-of-code-fellows/gsoc-2024-fellows.qmd

```diff
@@ -0,0 +1,294 @@
+
+**3. Testing and Development setup on my local computer**
+ 
+To get a proper environment for functionality creation and concurrent testing we required a proper testing setup so that we could test the new functions made at the same time. This was done using [Revise.jl](https://timholy.github.io/Revise.jl/stable/). My mentor helped me set it up, added Revise.jl to the global Julia environment and made a Julia script by the name “startup.jl” and added it to “/home/jay-sanjay/.julia/config/” path in my local computer.
+
+Here is the sample for the startup.jl file:
+```julia
+
+using BenchmarkTools
+using OhMyREPL
+using PackageCompatUI
+using Pkg
+using PkgTemplates
+using Revise
+using REPL
+
+###################################
+# GENERAL CONFIGURATION
+###################################
+ENV["JULIA_EDITOR"] = "nvim"
+
+###################################
+# Revise.jl
+###################################
+enable_autocomplete_brackets(false)
+
+###################################
+# HELPER FUNCTIONS
+###################################
+function template()
+    Template(;
+        user="jay-sanjay",
+        dir="~/FOSS",
+        authors="jaysanjay <jaysanjay@gmail.com> and contributors",
+        julia=v"1.6",
+        plugins=[
+            ProjectFile(; version=v"0.0.1"),
+            Git(),
+            Readme(),
+            License(; name="MIT"),
+            GitHubActions(; extra_versions=["1.6", "1", "nightly"]),
+            TagBot(),
+            Codecov(),
+            Documenter{GitHubActions}(),
+            Citation(; readme = true),
+            RegisterAction(),
+            BlueStyleBadge(),
+            Formatter(;style = "blue")
+        ],
+    )
+end
+
+```
+
+
+**4. Selecting treatments of interest**
+
+So, as a part of this, we used the previously mentioned research paper and discussion with the mentors we came up with logic for it. The first thing to do was to determine the moment in time from which selected treatments of interest should be included in the treatment pathway. The default is all treatments starting after the index date of the target cohort. For example, for a target cohort consisting of newly diagnosed patients, treatments after the moment of first diagnosis are included. However, it would also be desirable to include (some) treatments prior to the index date, for instance in case a specific disease diagnosis is only confirmed after initiating treatment. Therefore, periodPriorToIndex specifies the period (i.e. number of days) prior to the index date from which treatments should be included. We have created two dispatches for this function.
```

Comment:
It would be great to mention that you used FunSQL.jl here in some way with these implementations. Maybe a code snippet or two?